crates/contract/tests/sandbox/upgrade_to_current_contract.rs

```diff
@@ -261,3 +281,120 @@ async fn all_participants_get_valid_mock_attestation_for_soft_launch_upgrade() -
     );
     Ok(())
 }
+
+//// Verifies that upgrading the contract preserves state and allows the new
+/// functionality, in this case only CKD
+///
+/// This test:
+/// 1. Deploys an older version of the contract.
+/// 2. Initializes it with participants and submits a parameter update proposal.
+/// 3. Adds multiple domains with both `Ed25519` and `Secp256k1` schemes.
+/// 4. Submits pending signature requests across those domains.
+/// 5. Captures the full pre-upgrade state.
+/// 6. Upgrades the contract to the new version and runs `migrate()`.
+/// 7. Asserts that the state (participants, domains, proposals, signature requests, etc.)
+///    is identical post-upgrade.
+/// 10. Adds new domains, including CKD
+/// 11. Submits new signature and ckd requests
+/// 12. Confirms that pending signature and ckd requests created before and after the upgrade
+///    can still be responded to.
+#[rstest]
+#[tokio::test]
+async fn upgrade_allows_new_request_types(
```

Comment:
Is seems like this is an extension of `upgrade_preserves_state_and_requests`. It's nice to have a comprehensive test like this, but I would suggest to test the second part in isolation. There should be a `test_contract_sign_request_ckd` in the `sign.rs` file.

It might be nice to extract logic used by different tests into separate functions.

Reply:
Here what I was testing was exactly that the new contract with the old state can handle new request types. The second part was already tested elsewhere. If you think this test is not necessary we could simply remove it.